examples/touch/main.go

```diff
@@ -0,0 +1,253 @@
+		}
+	}
+
+	// What touches are new in this frame?
+	for _, id := range inpututil.JustPressedTouchIDs() {
+		x, y := ebiten.TouchPosition(id)
+		g.touches[ebiten.TouchID(id)] = &touch{
+			originX: x, originY: y,
+			currX: x, currY: y,
+		}
+	}
+
+	// Update the current position and durations of any touches that have
+	// neither begun nor ended in this frame.
+	for _, id := range ebiten.TouchIDs() {
+		t := g.touches[id]
+		t.duration = inpututil.TouchPressDuration(id)
+		t.currX, t.currY = ebiten.TouchPosition(id)
+	}
+
+	// Interpret the raw touch data that's been collected into g.touches into
+	// gestures like two-finger pinch or single-finger pan.
+	if len(g.touches) == 2 {
+		// Potentially the user is making a pinch gesture with two fingers.
+		// If the diff between their origins is different to the diff between
+		// their currents and if these two are not already a pinch, then this is
+		// a new pinch!
+		id1, id2 := ebiten.TouchIDs()[0], ebiten.TouchIDs()[1]
+		t1, t2 := g.touches[id1], g.touches[id2]
+		originDiff := distance(t1.originX, t1.originY, t2.originX, t2.originY)
+		currDiff := distance(t1.currX, t1.currY, t2.currX, t2.currY)
+		if g.pinch == nil && g.pan == nil && math.Abs(originDiff-currDiff) > 3 {
+			t1.wasPinch = true
+			t2.wasPinch = true
+			g.pinch = &pinch{
+				id1:     id1,
+				id2:     id2,
+				originH: originDiff,
+				prevH:   originDiff,
+			}
+		}
+	} else if len(g.touches) == 1 {
+		// Potentially this is a new pan.
+		id := ebiten.TouchIDs()[0]
+		t := g.touches[id]
+		if !t.wasPinch && g.pan == nil && g.pinch == nil {
+			diff := math.Abs(distance(t.originX, t.originY, t.currX, t.currY))
+			if diff > 1 {
+				t.isPan = true
+				g.pan = &pan{
+					id:      id,
+					originX: t.originX,
+					originY: t.originY,
+					prevX:   t.originX,
+					prevY:   t.originY,
+				}
+			}
+		}
+	}
+
+	// Copy any active pinch gesture's movement to the Game's zoom.
+	if g.pinch != nil {
+		x1, y1 := ebiten.TouchPosition(g.pinch.id1)
+		x2, y2 := ebiten.TouchPosition(g.pinch.id2)
+		curr := distance(x1, y1, x2, y2)
+		delta := curr - g.pinch.prevH
+		g.pinch.prevH = curr
+
+		g.zoom += (delta / 100) * g.zoom
+		if g.zoom < 0.25 {
+			g.zoom = 0.25
+		} else if g.zoom > 10 {
+			g.zoom = 10
+		}
+	}
+
+	// Copy any active pan gesture's movement to the Game's x and y pan values.
+	if g.pan != nil {
+		currX, currY := ebiten.TouchPosition(g.pan.id)
+		deltaX, deltaY := currX-g.pan.prevX, currY-g.pan.prevY
+
+		g.pan.prevX, g.pan.prevY = currX, currY
+
+		g.x += float64(deltaX)
+		g.y += float64(deltaY)
+	}
+
+	// If the user has tapped, then reset the Game's pan and zoom.
+	if len(g.taps) > 0 {
+		g.x = screenWidth / 2
+		g.y = screenHeight / 2
+		g.zoom = 1.0
+	}
+	return nil
+}
+
+func (g *Game) Draw(screen *ebiten.Image) {
+	op := &ebiten.DrawImageOptions{}
+
+	// Apply zoom.
+	op.GeoM.Scale(g.zoom, g.zoom)
+
+	// Apply pan.
+	op.GeoM.Translate(g.x, g.y)
+
+	// Centre the image (corrected by the current zoom).
```

Comment:
```suggestion
	// Center the image (corrected by the current zoom).
```
Oops, I'm assuming we should stick to US English?

Reply:
Yes, we use US English in this project.